### ISSUE TITLE
About Training Implenmentation detail of yolact

### ISSUE THREAD

**Issue by txytju** [dbolya/yolact#14]
Thanks for sharing your your great work!
I compared yolact's training config with that of retinanet since yolact is based on retinanet(I think) 
I have a few questions about the training config of yolact.
(1) the batch size on one GPU is 8, so how many GPUs did you use when training? 4 or 8? which means that total batch size is 32 or 64.  Retinanet's batch size is 16.
(2) the iterations is 800k, which is almost 10x larger than retinanet. why?
(3) the learning rate is 1e-3, which is 10 times smaller than retinanet, why?

Thanks!

**Comment by dbolya** [comment]
We're not exactly based on RetinaNet, but instead we based on an SSD codebase that we kind of retrofitted RetinaNet back into, so it's not perfect. For instance, we do the quite harsh data augmentation that SSD does, while if I remember correctly, RetinaNet only does horizontal flip. We also use a fixed image size of 550x550 without preserving aspect ratio like what SSD does, which is also different from the "scale the shorter side to be x" that IIRC RetinaNet and other modern detectors use.

These things affect training time and learning rate, so that's why our training parameters are very different. We are currently working to remove the dependency on data augmentation and to train and test with images of many different sizes, which should fix these issues. Hopefully doing that will reduce the need for an absurd 800k iterations (though this number is unoptimized, and could likely be already improved).

Also, we train with only one GPU (so total batch size is 8) more because of a lack of resources than anything. If you trained with 4-8 GPUs, you could reduce the number of iterations to 200k-100k (see #8).
